File: scripts/insert-mock-data.py
import requests
import random
from datetime import datetime, timedelta

API_URL = "http://localhost:8080/api.php"
NUM_TRADES = 100  # Change this number as needed

symbols = ['BTCUSD', 'ETHUSD', 'XRPUSD', 'LTCUSD', 'ADAUSD']
timeframes = ['1H', '4H', '1D', '1W']
position_types = ['Long', 'Short']

def random_date(start_date, end_date):
    time_between = end_date - start_date
    random_days = random.randrange(time_between.days)
    return start_date + timedelta(days=random_days)

def create_trade():
    try:
        # Generate random trade data
        trade_date = random_date(
            start_date=datetime(2023, 1, 1),
            end_date=datetime.now()
        ).strftime('%Y-%m-%d')
        
        payload = {
            "symbol": random.choice(symbols),
            "profit_loss": round(random.uniform(-500, 1000), 2),
            "trade_date": trade_date,
            "timeframe": random.choice(timeframes),
            "position_type": random.choice(position_types)
        }

        # Send POST request
        response = requests.post(
            API_URL,
            json=payload,
            headers={'Content-Type': 'application/json'}
        )
        
        if response.status_code != 200:
            print(f"\nError creating trade: {response.text}")
            
    except Exception as e:
        print(f"\nError: {str(e)}")

if __name__ == "__main__":
    print(f"Creating {NUM_TRADES} mock trades...")
    
    for i in range(NUM_TRADES):
        create_trade()
        print(f"\rProgress: {(i+1)/NUM_TRADES*100:.1f}%", end='', flush=True)
    
    print("\nDone! Check your dashboard.php")

# To test manually:
# curl -X POST http://localhost:8080/api.php -H "Content-Type: application/json" -d '{"symbol":"BTCUSD","profit_loss":250.5,"trade_date":"2024-03-01","timeframe":"1H","position_type":"Long"}'

#  curl -X POST http://localhost:8080/api.php \
#   -H "Content-Type: application/json" \
#   -d '{
#     "symbol": "BTCUSD",
#     "profit_loss": 250.50,
#     "trade_date": "2024-03-01",
#     "timeframe": "1H",
#     "position_type": "Long"
#   }'

File: scripts/insert-mock-data.js
import axios from 'axios';
import * as d3 from 'd3';

const API_URL = process.env.API_URL || 'http://localhost:8080/api.php';

const symbols = ['BTCUSD', 'ETHUSD', 'XRPUSD', 'LTCUSD', 'ADAUSD'];
const timeframes = ['1H', '4H', '1D', '1W'];
const positionTypes = ['Long', 'Short'];

function randomDate(start, end) {
  return new Date(start.getTime() + Math.random() * (end.getTime() - start.getTime()));
}

async function createMockTrade() {
  const tradeDate = randomDate(new Date(2023, 0, 1), new Date());
  const profitLoss = d3.randomUniform(-500, 1000)();

  try {
    await axios.post(API_URL, {
      symbol: symbols[Math.floor(Math.random() * symbols.length)],
      profit_loss: profitLoss.toFixed(2),
      trade_date: tradeDate.toISOString().split('T')[0],
      timeframe: timeframes[Math.floor(Math.random() * timeframes.length)],
      position_type: positionTypes[Math.floor(Math.random() * positionTypes.length)]
    });
  } catch (error) {
    console.error('Error creating trade:', error.message);
  }
}

async function populateData(numEntries = 100) {
  console.log(`Inserting ${numEntries} mock trades...`);
  
  for (let i = 0; i < numEntries; i++) {
    await createMockTrade();
    process.stdout.write(`\rProgress: ${((i + 1) / numEntries * 100).toFixed(1)}%`);
  }
  
  console.log('\nData population completed!');
}

// Start population
populateData(100).catch(console.error);